citesphere-model/src/main/java/edu/asu/diging/citesphere/data/bib/impl/CitationMongoDao.java

```diff
@@ -72,4 +88,75 @@ public List<? extends ICitation> findCitationsInCollection(String groupId, Strin
         return mongoTemplate.find(query, Citation.class);
     }
     
+    @Override
+    public Persons findAllPeople(String groupId, long start, int pageSize) {
+        MatchOperation match = Aggregation.match(Criteria.where("group").is(groupId));
+
+        ProjectionOperation project = Aggregation.project().and("group").as("_id").and("authors")
+                .unionArrays("editors", "otherCreators.person").as("persons").and("key").as("citationKey");
+
+        UnwindOperation unwind = Aggregation.unwind("persons");
+
+        GroupOperation group = Aggregation.group("_id").addToSet(new BasicDBObject() {
+            {
+                put("name", "$persons.name");
+                put("firstName", "$persons.firstName");
+                put("lastName", "$persons.lastName");
+                put("uri", "$persons.uri");
+                put("citationKey", "$citationKey");
+            }
+        }).as("persons");
+
+        ProjectionOperation projectResultCount = Aggregation.project().and("_id").as("_id").and("persons")
+                .as("persons").and("persons").size().as("totalResults");
+        
+        UnwindOperation unwind1 = Aggregation.unwind("persons");
+
+        SortOperation sort = Aggregation.sort(Direction.ASC, "persons.name");
+        
+        SkipOperation skip = Aggregation.skip(start);
+        
+        LimitOperation limit = Aggregation.limit(pageSize);
+
+        GroupOperation groupAfterSort = Aggregation.group("_id").push("persons").as("persons")
+                .first("totalResults").as("totalResults");
+
+        AggregationResults<Persons> result = mongoTemplate.aggregate(
+                Aggregation.newAggregation(match, project, unwind, group, projectResultCount, unwind1, sort, skip, limit, groupAfterSort),
+                Citation.class, Persons.class);
+
+        return result.getUniqueMappedResult();
```

Comment:
and shouldn't the people be uniquely identified by uri? which doesn't seem to happen here.

Reply:
I think this woulld give the right person list, does that result in the same?
```
[{$match: {
  "group": "2601560"
}}, {$addFields: {
  "persons": {$setUnion: ["$authors", "$editors", "$otherCreators"]}
}}, {$unwind: {
  path: "$persons"
}}, {$addFields: {
  "ident": { 
    $cond: {
      if: {$ne: ["$persons.uri", ""]}, 
      then: "$persons.uri", 
      else: {$concat: ["$key","$persons.name"]}
    }
  }
}
}, {$group: {
  _id: "$ident",
  persons: {
    $push: "$persons"
  }
}}]
```
This does assume that iff there are two people with exactly the same name and no URI in the same citation, its the same person (which should be fine as long as they would be different people across citations).